### PR TITLE
Filter media upload on allowed formats

### PR DESCRIPTION
**Pull request by ngilain**
https://github.com/open-orchestra/open-orchestra-media-admin-bundle/pull/86
https://github.com/open-orchestra/open-orchestra-media-bundle/pull/135

### CODE REVIEW

File: MediaAdminBundle/Controller/Api/MediaController.php [hunk 1]
```diff
@@ -101,10 +101,10 @@ public function deleteAction($mediaId)
      * 
      * @Config\Route("/upload/{folderId}", name="open_orchestra_api_media_upload")
      * Config\Method({"POST"})
-     *
+     *
```
Comment (nicolasThal) [inline MediaAdminBundle/Controller/Api/MediaController.php]: white space to suppress

File: MediaAdminBundle/Controller/Api/MediaController.php [hunk 2]
```diff
@@ -101,10 +101,10 @@ public function deleteAction($mediaId)
      * 
      * @Config\Route("/upload/{folderId}", name="open_orchestra_api_media_upload")
      * Config\Method({"POST"})
-     *
+     * 
      * @Config\Security("is_granted('ROLE_ACCESS_CREATE_MEDIA')")
-     *
-     * @return FacadeInterface|array
+     *
```
Comment (nicolasThal) [inline MediaAdminBundle/Controller/Api/MediaController.php]: same

File: MediaAdminBundle/Controller/Api/MediaController.php [hunk 3]
```diff
@@ -113,13 +113,22 @@ public function uploadAction($folderId, Request $request)
...
+                    ->createMediaFromUploadedFile($uploadedFile, $filename, $folderId);
+
+                return $this->get('open_orchestra_api.transformer_manager')
+                    ->get('media')->transform($media);
+            } else {
+                $translator = $this->container->get('translator');
 
-            return $this->get('open_orchestra_api.transformer_manager')
-                ->get('media')->transform($media);
+                return new Response(
```
Comment (nicolasThal) [inline MediaAdminBundle/Controller/Api/MediaController.php]: As the Api\Serialize annotation as been put on the full controller, you cannot return a Response, you should either : 
- put the annotation on each controller action
- return an empty array or a facade

Comment (nicolasThal) [inline MediaAdminBundle/Controller/Api/MediaController.php]: Or you could also throw an exception

Comment (ngilain) [inline MediaAdminBundle/Controller/Api/MediaController.php]: @nicolasThal : I tried to throws exceptions but it breaks the flow.js treatment of the return, instead returning a 403 or 202 response goes fine.
The Api\Serialize annotation is only read if the KernelEvents::VIEW is fired. This is only the case when a non Response value is return by a controller. So when I return a custom Response in this controller, the serialize mechanism is even not called and that's why it works.

**Comment by ngilain** [comment]
@nicolasThal : update done

File: MediaAdminBundle/Controller/Api/MediaController.php [hunk 4]
```diff
@@ -113,13 +113,22 @@ public function uploadAction($folderId, Request $request)
...
-                ->createMediaFromUploadedFile($uploadedFile, $filename, $folderId);
+            if ($saveMediaManager->isFileAllowed($filename)) {
+                $media = $this->container->get('open_orchestra_media.manager.media')
+                    ->createMediaFromUploadedFile($uploadedFile, $filename, $folderId);
 
-            return $this->get('open_orchestra_api.transformer_manager')
-                ->get('media')->transform($media);
+                return $this->get('open_orchestra_api.transformer_manager')
+                    ->get('media')->transform($media);
+            } else {
```
Comment (nicolasThal) [inline MediaAdminBundle/Controller/Api/MediaController.php]: The `else` is not usefull in this case

File: MediaAdminBundle/Controller/Api/MediaController.php (hunk 3 above)
Comment (nicolasThal) [inline MediaAdminBundle/Controller/Api/MediaController.php]: All right.

I still think it is not really right, but as you should make it work with flow.js I will agree to let it go

**Comment by nicolasThal** [comment]
:+1: with the `else` suppressed

**Pull request merged**